### ISSUE TITLE
Async streams provide a better way  ...but without progress

### ISSUE THREAD

**Issue by f3** [dotnet/docs#36515]
Good document, and _almost_ thorough 😁

The conversion in the _[Async streams provide a better way](https://learn.microsoft.com/en-us/dotnet/csharp/asynchronous-programming/generate-consume-asynchronous-stream#async-streams-provide-a-better-way)_ section totally strip out the use of `IProgress<T>` from the method converted to use async streams.  Can't wrap my head around whether that's even "do-able" (I suspect not), but it's not mentioned in the doc that you lose the ability to use a `Progress<T>`...  Meh, I'm over it 😉

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: e2d2f1e2-6b7f-d696-0e67-441c1f4bcfe9
* Version Independent ID: 0910174f-c065-5cb3-3c1a-7d44409903d2
* Content: [Generate and consume async streams](https://learn.microsoft.com/en-us/dotnet/csharp/asynchronous-programming/generate-consume-asynchronous-stream)
* Content Source: [docs/csharp/asynchronous-programming/generate-consume-asynchronous-stream.md](https://github.com/dotnet/docs/blob/main/docs/csharp/asynchronous-programming/generate-consume-asynchronous-stream.md)
* Product: **dotnet-csharp**
* Technology: **csharp-async**
* GitHub Login: @BillWagner
* Microsoft Alias: **wiwagn**

**Comment by BillWagner** [comment]
Hi @f3 

This is a good suggestion. We should call it out more. Using async streams, you don't need a progress callback in order to report progress. The initial code doesn't return any of the elements from the web request until *all* elements have been retrieved, using multiple web requests. When the code is structured that way, it's often useful to have a progress callback to indicate that *some* results have been retrieved, and the API is still working to retrieve more.

In the async enumerable version, as soon as the first web request returns, the client using the async enumerable can begin processing those results. When you run the sample, the output illustrates this well: The initial version writes several messages from the progress callback before any of the results are printed. In the async enumerable version, as soon as the first results are available, the messages show the first results.

I've added this to our backlog to update when we next update this article. I've also added the help wanted label if anyone wants to submit a PR from these notes before we make the next update.
